### PR TITLE
Feature/improved final standings

### LINKED ISSUE #138: As a Participant I want to see more details on the Final Standings
Currently the participant just sees the scores in the Final Standings page. 
Because of bonus points it is difficult to understand how your points are calculated and even how many you got right. 

Participant would like to see:
- The answers they gave to each question,  the correct answer,  whther they were marked correct or not,  who was fastest (if its them it should say +1 to indicate they got bonus point.      This should allow them to understand how everything was calculated
-  In the final standings table it would be good to see how many bonus points the other contestants got to understand better the scores  e.g.   Score:   7  (5 + 2) 

### Acceptance Criteria

QuizMaster clicks the show FInal Standings button
Quiz Participants see the Final Standings and on the same page see the breakdown of results as described above.   The list of answers only needs to be THEIR OWN answers.

### PR DESCRIPTION
**Pull request by richielewis**
Added more information to Final Standings table. 

Also added a Questions summary table below Final Standings so participants can see the answers they gave and how it compared to real answer.  

API changes involve 

- New API to update ContestantAnswer after quiz master marks answers
- New API to retrieve final summary information which only returns info if quiz has state QuizEnded

Resolves #138 

See screenshot 
image

### CODE REVIEW

File: quizmaster-client-app/src/components/QuizHosting/QuizStandings.tsx [hunk 1]
```diff
@@ -8,36 +8,177 @@ import TableHead from "@material-ui/core/TableHead";
...
+          currentScore = contestant.score;
+          newRankings[contestant.id] = currentRank;
+          currentRankCount = 0;
+        }
+      });
+    setRankings(newRankings);
+  }, [props.contestantStandings]);
+
+  const getNameClass = (id: string) => {
+    if (id == participantId) {
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/QuizHosting/QuizStandings.tsx:93]: could make these ternary, e.g.:
return id == participantId ?  classes.ownName : classes.otherName;

**Review by calumtomeny — COMMENTED** [review]
Very nice addition.

File: quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx [hunk 2]
```diff
@@ -247,18 +259,32 @@ export default function QuizHoster() {
...
     });
-    axios.post(`/api/contestants/updates`, updateCommands).then(() => {
-      setShowQuizMarker(false);
-    });
+    const answersUpdateCommand = {
+      QuestionNo: currentQuestionNumber,
+      AnswerUpdates: roundScores,
+    };
+    axios
+      .post(`/api/quizzes/${id}/updatecontestantanswers`, answersUpdateCommand)
```
[the remark this answers is not in the document]
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx:282]: could a POST also mean adding new ones?

**Review by calumtomeny — DISMISSED** [review]
Nice work - looking forward to trying it out.

File: quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx:282]: Don't see an advantage to not having explicitly named endpoints

File: quizmaster-client-app/src/components/QuizHosting/QuizStandings.tsx (hunk 1 above)
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizHosting/QuizStandings.tsx]: Done

File: QuizMaster.Application/Contestants/FinalSummary.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+            public async Task<List<QuestionSummary>> Handle(Query request, CancellationToken cancellationToken)
+            {
+                var contestant = await context.Contestants.Include(x => x.ContestantAnswers).SingleOrDefaultAsync(x => x.Id == request.Id);
+                if (contestant == null)
+                {
+                    return null;
+                }
+
+                var quiz = await context.Quiz.Include("QuizQuestions.ContestantAnswers").SingleOrDefaultAsync(x => x.Id == contestant.QuizId);
```
[the remark this answers is not in the document]
Comment (richielewis) [reply QuizMaster.Application/Contestants/FinalSummary.cs]: Done

**Review by timloudon — APPROVED** [review]
Awesome work Rich

**Pull request merged**